contracts/src/v0.1/mocks/L2VRFConsumerMock.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import "../VRFConsumerBase.sol";
+import "../interfaces/IL2Endpoint.sol";
+
+contract L2VRFConsumerMock is VRFConsumerBase {
+    uint256 public sRandomWord;
+    address private sOwner;
+
+    IL2Endpoint L2ENDPOINT;
+
+    error OnlyOwner(address notOwner);
+
+    modifier onlyOwner() {
+        if (msg.sender != sOwner) {
+            revert OnlyOwner(msg.sender);
+        }
+        _;
+    }
+
+    constructor(address l2Endpoint) VRFConsumerBase(l2Endpoint) {
+        sOwner = msg.sender;
+        L2ENDPOINT = IL2Endpoint(l2Endpoint);
+    }
+
+    // Receive remaining payment from requestRandomWordsPayment
+    receive() external payable {}
+
+    function requestRandomWords(
+        bytes32 keyHash,
+        uint64 accId,
+        uint32 callbackGasLimit,
+        uint32 numWords
+    ) public onlyOwner returns (uint256 requestId) {
+        requestId = L2ENDPOINT.requestRandomWords(keyHash, accId, callbackGasLimit, numWords);
```

Comment:
We are not able to gauge whether account has enough KLAY when observing from L2. However, we might be able to at least check whether somebody owns certain account if we have create `account creation` triggered by transaction on L2. Please think about this approach. It could help us avoid some issues.

Reply:
Let me think about this. I believe it will bring more complications to the current implementations.

Reply:
@KelvinThai Yes! Please just make an issue from it so we don't forget about it 🙏